app/models/miq_server/worker_management/monitor.rb

```diff
@@ -26,7 +26,18 @@ def monitor_workers
     processed_worker_ids += check_not_responding
     processed_worker_ids += check_pending_stop
     processed_worker_ids += clean_worker_records
-    processed_worker_ids += request_workers_to_sync_config(resync_needed)
+
+    # Monitor all remaining current worker records
+    processed_worker_ids += miq_workers.where(:status => MiqWorker::STATUSES_CURRENT_OR_STARTING).each do |worker|
+      # Check their queue messages for timeout
+      worker.validate_active_messages
+      # Push the heartbeat into the database
+      persist_last_heartbeat(worker)
+      # Check the worker record for heartbeat timeouts
+      next unless validate_worker(worker)
+      # Tell the valid workers to sync config if needed
+      worker_set_message(worker, "sync_config") if resync_needed
+    end.map(&:id)
```

Comment:
I think something is wrong with this since `.each .map` seems wrong.  Did you mean `.select .map`?  Otherwise you end up with all the ids regardless of what's done in the block.

Reply:
Nope, this is what I wanted. It turns out that the only gate to adding the worker id to the processed worker list was that it was current or starting, so I just added that to the query.

I think this should be nearly identical to `request_workers_to_sync_config` in effect.

Reply:
ok cool...just wanted to make sure.